java/oauth-custom-challengehandler/src/main/java/com/esri/arcgisruntime/sample/oauthcustomchallengehandler/MainActivity.java

```diff
@@ -0,0 +1,304 @@
+            mOAuthLoginDialog.show();
+          }
+        });
+
+        try {
+          // Wait for the OAuth browser page to be shown and user to login and for the CustomWebViewClient
+          // to get the auth code and request an OAuthTokenCredential with it
+          oAuthChallengeCountDownLatch.await();
+        } catch (InterruptedException ie) {
+        }
+        // Get the credential from the latch once it has been counted down
+        OAuthTokenCredential credential = oAuthChallengeCountDownLatch.getOAuthTokenCredential();
+        if (credential != null) {
+          ret = new AuthenticationChallengeResponse(AuthenticationChallengeResponse.Action.CONTINUE_WITH_CREDENTIAL,
+              credential);
+        }
+        // dismiss the dialog
+        mOAuthLoginDialog.dismiss();
+      }
+      return ret != null ? ret : new AuthenticationChallengeResponse(AuthenticationChallengeResponse.Action.CANCEL, challenge);
+    }
+  };
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_main);
+
+    // set custom challenge handler on the AuthenticationManager
+    AuthenticationManager.setAuthenticationChallengeHandler(mCustomAuthenticationChallengeHandler);
+
+    // initialize the dialog to display the OAuth login page
+    mOAuthLoginDialog = new Dialog(this);
+
+    // inflate MapView from layout
+    mMapView = (MapView) findViewById(R.id.mapView);
+
+    // add the OAuth configuration to the AuthenticationManager
+    try {
+      OAuthConfiguration config = new OAuthConfiguration(getString(R.string.portal_url), getString(R.string.client_id), getString(R.string.redirect_url));
+      AuthenticationManager.addOAuthConfiguration(config);
+    } catch (MalformedURLException e) {
+      throw new IllegalArgumentException(e.getMessage(), e.getCause());
+    }
+
+    // create a portal instance for the portal to load
+    Portal portal = new Portal(getString(R.string.portal_url));
+
+    // create a portal item with the itemId of the web map
+    PortalItem webMapItem = new PortalItem(portal, getString(R.string.webmap_id));
+
+    // create a map with the portal item
+    ArcGISMap map = new ArcGISMap(webMapItem);
+
+    // set the map to the map view
+    mMapView.setMap(map);
+  }
+
+  @Override
+  protected void onResume() {
+    super.onResume();
+    mMapView.resume();
+  }
+
+  @Override
+  protected void onPause() {
+    super.onPause();
+    mMapView.pause();
+  }
+
+  @Override
+  protected void onDestroy() {
+    super.onDestroy();
+    mMapView.dispose();
+  }
+
+  /**
+   * CustomWebViewClient overrides the shouldOverrideUrlLoading method to handle errors on the login page and to request
+   * OAuthTokenCredential after having received the auth code when the user provides the login credential/
+   */
+  private class CustomWebViewClient extends WebViewClient {
+
+    private final String mPortal;
+
```

Comment:
can remove the whitespace between fields

Reply:
removed